### PR TITLE
dm: add meta binlog precheck

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/8999

### What is changed and how it works?
add meta binlog precheck

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
dm: add meta binlog position precheck
```

### CODE REVIEW

File: dm/pkg/checker/binlog.go [hunk 1]
```diff
@@ -262,3 +273,109 @@ func (c *BinlogDBChecker) Check(ctx context.Context) *Result {
...
+		streamer, err = syncer.StartSyncGTID(gtidSet)
+	} else {
+		streamer, err = syncer.StartSync(mysql.Position{Name: c.meta.BinLogName, Pos: c.meta.BinLogPos})
+	}
+	if err != nil {
+		markCheckError(result, err)
+		return result
+	}
+	// if we don't get a new event after 15s, it means there is no new event in the binlog
+	ctx2, cancel := context.WithTimeout(ctx, 15*time.Second)
```
Comment (lance6716) [inline dm/pkg/checker/binlog.go:366]: start-task will be blocked at least 15 seconds?

Comment (lichunzhu) [reply dm/pkg/checker/binlog.go:366]: In worst case, yes. Don't find a better way to fix this problem. Any advice?

Comment (lance6716) [reply dm/pkg/checker/binlog.go:366]: maybe we can check the position by SHOW BINARY LOGS or gtid_purged

File: dm/pkg/checker/binlog.go [hunk 2]
```diff
@@ -262,3 +273,109 @@ func (c *BinlogDBChecker) Check(ctx context.Context) *Result {
...
+		Flavor:    flavor,
+		Host:      c.sourceCfg.Host,
+		Port:      uint16(c.sourceCfg.Port),
+		User:      c.sourceCfg.User,
+		Password:  c.sourceCfg.Password,
+		TLSConfig: tlsConfig,
+		Localhost: h,
+	}
+
+	syncer := replication.NewBinlogSyncer(syncCfg)
```
Comment (lance6716) [inline dm/pkg/checker/binlog.go:349]: need to close syncer

**Review by lance6716 — APPROVED** [review]

**Review by okJiang — APPROVED** [review]
LGTM

File: dm/pkg/checker/binlog.go (hunk 2 above)
Comment (lichunzhu) [reply dm/pkg/checker/binlog.go:349]: fixed in https://github.com/pingcap/tiflow/pull/9000/commits/73fdfe365c13389a08aab07a7452cd475d64ad15

**Pull request merged**